spock-core/src/main/java/org/spockframework/mock/IStaticMockController.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.spockframework.mock;
+
+import org.spockframework.mock.runtime.IMockMaker;
+import org.spockframework.util.Beta;
+
+import java.util.concurrent.Callable;
+
+/**
+ * The {@code IStaticMockController} provides API to activate a static mocks on non-test {@link Thread Threads}.
+ *
+ * @since 2.4
+ */
+@Beta
+public interface IStaticMockController {
+  /**
+   * Runs the code with the static mocks activated on the current {@link Thread}.
+   *
+   * <p>Note: You only need this if your current {@code Thread} is not the test thread.
+   * On the test {@code Thread}, the static mocks is automatically activated.</p>
+   *
+   * @param code the code to execute
+   */
+  void runWithActiveStaticMocks(Runnable code);
```

Comment:
I would say this is not mockito impl details, but rather our API requirements documented in the `IMockMaker.IStaticMock`. So if another mock maker wants to contribute static mocking ability, it also needs to implement that in a thread-local way.

Reply:
Returning to the related `mockConstruction` case, AFAIU this would also require something similar to work on other threads. So, if we followed this we'd have sibling methods of `runWithActiveConstructorMocks` or something.

Maybe, it would be better to have something more generic, like `runWithThreadBoundMocks`/`runWithThreadAwareMocks`. Or just `runWithActiveMocks`, although that might suggest to use it even if it is not needed.

Reply:
Yes you are right, this should be more general. I will change that to `runWithThreadAwareMocks`.

Reply:
Done.